### PR TITLE
feat: add hybrid logical clock to codebase

### PR DESCRIPTION
**Pull request by ryardley**
Part of #1050 - read if you want to know why we need this in short this is so we can ask the lilbp2p network for events we have missed from a particular HLC onwards.

This adds a hybrid logical clock to manage our event sync but does not yet add it to our events.

Custom implementation is so that we can control the serialization structure. Feel free to bike shed on this - i thought about going to u64 but that limits the number of years the clock can handle. I think u128 is fine unless we really want to store this on contract (not the plan - see #1050) 

I have been concerned that in case we need to build a node in a different language we should control the format of the serialized clock so as to be able to specify it's structure to say implement it in go etc. which is why I did not choose to use uhlc

This is pretty straight foward basically concatenating big endian bytes together.

```text
┌────────────────┬────────────────┬────────────────┐
│   ts (u64)     │  counter (u32) │   node (u32)   │
│  bytes 0..8    │  bytes 8..12   │  bytes 12..16  │
└────────────────┴────────────────┴────────────────┘
```

I also chose to implement prop testing to make sure we cover corner cases.

## Summary by CodeRabbit

* **New Features**
  * Adds a public Hybrid Logical Clock (HLC) module for generating monotonic, mergeable timestamps with configurable drift tolerance and safe concurrent use.

* **Chores**
  * Updated dependencies to enable property-based testing and improved error handling support.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: crates/events/src/hlc.rs [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+        if let Some(next_counter) = new_counter.checked_add(1) {
+            inner.counter = next_counter;
+            inner.ts = max_ts;
+        } else {
+            // Counter overflow - advance timestamp and reset
+            inner.ts = max_ts
+                .checked_add(1)
+                .ok_or(HlcError::CounterOverflow { ts: max_ts })?;
+            inner.counter = 0;
+        }
```
[the remark this answers is not in the document]
Comment (ryardley) [reply crates/events/src/hlc.rs]: Fixed
